arch/risc-v/src/cmake/Toolchain.cmake

```diff
@@ -259,7 +259,16 @@ if(CONFIG_RISCV_TOOLCHAIN STREQUAL GNU_RVG)
   endif()
 
   if(CONFIG_ARCH_RV_ISA_ZICSR_ZIFENCEI)
-    set(ARCHCPUEXTFLAGS ${ARCHCPUEXTFLAGS}_zicsr_zifencei)
+    if(NOT DEFINED GCCVER)
+      execute_process(COMMAND ${CMAKE_CXX_COMPILER} --version
+                      OUTPUT_VARIABLE GCC_VERSION_OUTPUT)
+      string(REGEX MATCH "\\+\\+.* ([0-9]+)\\.[0-9]+" GCC_VERSION_REGEX
+                   "${GCC_VERSION_OUTPUT}")
+      set(GCCVER ${CMAKE_MATCH_1})
+    endif()
```

Comment:
@xuxin930 @xiaoxiang781216 Let's define the get gcc version as a function(cmake, marco in makefile) and remove the duplicate logic from other place

Reply:
ok

Reply:
let's change ARCH_RV_ISA_ZICSR_ZIFENCEI to n instead:
https://github.com/apache/nuttx/blob/master/arch/risc-v/Kconfig#L414-L427

Reply:
> @xuxin930 @xiaoxiang781216 Let's define the get gcc version as a function(cmake, marco in makefile) and remove the duplicate logic from other place

Sure, I'll complete this action.